quilt_repository_format.md

```diff
@@ -0,0 +1,137 @@
+# Quilt CLI Local Storage
+
+## Motivation
+The primary motivation for unifying the local package store is that any Python process will be able to access any locally installed Quilt package. A common point of friction for Quilt users has been that packages built on the command line aren't visible inside Jupyter notebooks if the Jupyter server is running in a different directory. Unifiying the local storage by adopting the Docker storage model solves that problem and also enables useful future features like comparing (diff) package versions and tracking revision history locally without pushing to a registry.
```

Comment:
can't accurately call this a common point of friction; but it is a point of friction per #164 and #166. not clear how many times this is the problem vs. number of times ipython is executing where it can't find quilt.

Reply:
@stillmatic, @salvor7 does this docker-like approach of having a single local package repository that is capable of maintaining multiple package instances simultaneously meet your needs?

Reply:
I think that having the base directory is useful. That would allow different Quilt clients (e.g. in R) to easily access the package root. This change also helps ensure that you are using the 'right' version of the package, versus having several copies in different folders.

In my specific use-case, a number of users share one storage system (everybody has their own folder under a single root folder, but no real restrictions on what you can do). Having a single Quilt folder rather than individual Quilt folders under each person's namespace would save on space and hassle, if we are reusing packages.

Reply:
We've talked about using a shared folder for quilt_packages and I left the hooks in for package lookups to check multiple places. Quick question, what would you want the permissions/security model to be for the shared folder? By default, anyone with access to that folder would be able to read the packages there.